src/user/controller/privileges.ts

```diff
@@ -0,0 +1,48 @@
+import Controller from '@curveball/controller';
+import { Context } from '@curveball/core';
+import { BadRequest } from '@curveball/http-errors';
+import * as privilegeService from '../../privilege/service';
+import { PrivilegeMap } from '../../privilege/types';
+import * as hal from '../formats/hal';
+import * as userService from '../service';
+// import * as groupService from '../../group/service';
+
+class UserEditPrivilegesController extends Controller {
+
+  async get(ctx: Context) {
+
+    const user = await userService.findById(+ctx.params.id);
+    const privileges = await privilegeService.getPrivilegesForPrincipal(user);
```

Comment:
The issue with this is that it doesn't just pull in a users' immediate privileges, but also all the privileges from any groups the user is in.

So I think we need a new function in the privilege service (like `getDirectPrivileges` or `getImmediatePrivileges`)